### ISSUE TITLE
[Core] Add __repr__ to ItemPaged

### ISSUE THREAD

**Issue by annatisch** [Azure/azure-sdk-for-python#8191]
Users instinctively try to print the response to a list operation, and the result is difficult to determine the iterable behaviour.

**Comment by lmazuel** [comment]
After teams discussion:
```
<iterator object azure.core.paging.ItemPaged at memory_address>
```

@annatisch @KieranBrantnerMagee @johanste for thumbs up/down.

You get my thumbs up :)

**Comment by xiangyan99** [comment]
fixed in #8678
